tests/python/unit/test_ontology_tagger.py

```diff
@@ -24,6 +26,7 @@ def test_list_sentences():
     assert isinstance(tagger, Tagger)
     text_df = pd.DataFrame({"text": [float("nan")]})
     text_column = "text"
+    text_df[text_column] = text_df[text_column].fillna("")
```

Comment:
Unit tests shouldn't copy code used by a given method, they should call the method directly.

This will avoid the common case where you may change a method in the future, but forget to update the unit test.

Now, it means you need a private method for this :)

Reply:
I've created a reusable method in plugin_io_utils instead : 1e16f3eb

Reply:
Sorry, I think there was a misunderstanding. I was not referring to this particular line of code. As a matter of fact, you don't need a function for a one-liner like fillna("").

 I was referring to the fact that overall, unit tests are not meant to test copies of the code in modules. A unit test should test the modules themselves.

You can have a look at these tests as an example: https://github.com/dataiku/dss-plugin-nlp-preparation/blob/main/tests/python/unit/test_spacy_tokenizer.py

Reply:
Aha yes there was a misunderstanding here. Anyway, if you look at the test right now, I think this is already the case. Tell me if I'm wrong :)

Reply:
No, the test function shouldn't do all the fillna, the spacy.load, etc. These steps should be handled by the tagger methods. 

Good unit test is about

```
def test_class_method():
    input = 42
    class_instance = MySuperClass(param="truc")
    output = class_instance.super_method(input)
    assert output = "expected"

```

Reply:
Sorry I should have explain that you shouldn't look at this commit. I put this commit in the comment to refer about "fillna" method. But I did change the unit_test elsewhere, 1e16f3e 74a8d5b. 
Just look at the file from files changed  view, you'll see how it looks like now.

Reply:
Ah yes indeed that's better!
For next time, perhaps keep all changes to one aspect of a thread to a single commit. Will be easier for me or another reviewer to follow.